os/arch/arm/src/amebasmart/amebasmart_i2s.c

```diff
@@ -1785,11 +1787,17 @@ struct i2s_dev_s *amebasmart_i2s_initialize(uint16_t port, bool is_reinit)
 	int ret;
 
 	/* Assign HW configuration */
+#ifdef CONFIG_AMEBASMART_I2S2
 	if (port == I2S_NUM_2) {
 		hw_config_s = (struct amebasmart_i2s_config_s *)&amebasmart_i2s2_config;
-	} else if (port == I2S_NUM_3) {
+	} else
+#endif
+#ifdef CONFIG_AMEBASMART_I2S3
+	if (port == I2S_NUM_3) {
 		hw_config_s = (struct amebasmart_i2s_config_s *)&amebasmart_i2s3_config;
-	} else {
+	} else 
+#endif
+	{
 		i2serr("Please select I2S2 or I2S3 bus\n");
 		return NULL;
```

Comment:
This looks strange. Is this function only for I2S2 and I2S3?
Don't you call this with I2S0 or I2S1?
If you do, this shows wrong error message.

Reply:
As i check the history, 
Because I2S0/1 are for internal codec, that is why I2S2/3 are used. 
I2S2 for TX and I2S3 for RX. 
Error message will be added if I2S0/1 is used.